src/index.js

```diff
@@ -11,11 +13,6 @@ const { multiaddrToNetConfig } = require('./utils')
 const { AbortError } = require('abortable-iterator')
 const { CODE_CIRCUIT, CODE_P2P } = require('./constants')
 
-/**
- * @typedef {import('multiaddr').Multiaddr} Multiaddr
- * @typedef {import('libp2p-interfaces/src/connection').Connection} Connection
```

Comment:
We are using the Connection/Multiaddr types below, it needs to be specified here. We will also need to specify the upgrader from libp2p https://github.com/libp2p/js-libp2p/blob/master/src/upgrader.js